src/Exception/ApiError.php

```diff
@@ -108,18 +110,23 @@ public static function throwFromResponse(ResponseInterface $response): void
             case self::INTERNAL_ERROR:
                 throw new ApiError($response, $details);
             case self::AUTHORIZATION_FAILED:
+            case self::INVALID_TOKEN:
                 throw new AuthFailed($response, $details);
             case self::DUPLICATE_DATA:
                 throw new DuplicateData($response, $details);
             case self::LIMIT_EXCEEDED:
                 throw new LimitExceeded($response, $details);
         }
+
+        if (strlen((string)$response->getBody()) <= 0) {
+            throw new ApiError($response, $details);
+        }
     }
 
     protected static function getErrorCodeAndDetailsFromResponse(ResponseInterface $response): array
     {
         try {
-            $body = json_decode($response->getBody(), true);
+            $body = json_decode((string)$response->getBody(), true);
```

Comment:
This could be `$response->getBody()->getContents()`

I'm mixed as I also do (string) sometimes so I dont know whats better

Reply:
i'm used to both so for now i'll leave it at that